pysat/_params.py

```diff
@@ -0,0 +1,319 @@
+                 before giving up.
+            'user_modules' : {}; Stores information on modules registered within
+                pysat. Used by `pysat.utils.registry`
+            'warn_empty_file_list' : False; Raises a warning if no files are
+                found for a given pysat.Instrument.
+
+    Stored pysat parameters without a working default value:
+        'data_dirs': Stores locations of top-level directories pysat uses to
+            store and load data.
+
+    """
+
+    def __init__(self, path=None, create_new=False):
+        sfname = 'pysat_settings.json'
+        self.data = {}
+        self.file_path = None
+
+        # Define default parameters and values
+        dir_format = os.path.join('{platform}', '{name}', '{tag}', '{inst_id}')
+        defaults = {'clean_level': 'clean',
+                    'directory_format': dir_format,
+                    'ignore_empty_files': False,
+                    'file_timeout': 10,
+                    'update_files': True,
+                    'user_modules': {},
+                    'warn_empty_file_list': False}
+
+        # Attach default parameters and values to object
+        self.defaults = defaults
+
+        # Define stored pysat parameters without a default setting
+        non_defaults = ['data_dirs']
+        self.non_defaults = non_defaults
+
+        # If path provided, use it. Otherwise, iterate through potential
+        # locations until file is found.
+        if path is not None:
+            # Confirm path is valid
+            if not os.path.exists(path):
+                estr = ''.join(('Supplied path does not exist on the local ',
+                                'system. Please create it and try again.'))
+                raise RuntimeError(estr)
+
+            # Store full file path including fixed settings file name
+            self.file_path = os.path.join(path, sfname)
+
+        else:
+            # Cycle through locations and load any pysat parameter files found.
+            # First, check current working directory, then pysat user directory.
+            # Stop looking once an existing file is found.
+            file_locs = [os.path.join('.', sfname),
+                         os.path.join(os.path.expanduser('~'), '.pysat',
+                                      sfname)]
+            for fileloc in file_locs:
+                if os.path.isfile(fileloc):
+                    self.file_path = fileloc
+                    break
+
+            # Ensure we have a valid file if the user isn't creating a new one.
+            if self.file_path is None and (not create_new):
+                estr = ''.join(('pysat is unable to locate a user settings ',
+                                'file. Please check the locations, "./" or ',
+                                '"~/.pysat" for the file "pysat_settings.json"',
+                                '.'))
+                raise RuntimeError(estr)
```

Comment:
Should this be OSError?

Reply:
Still waiting for a response on this one.

Reply:
I missed that one somehow. I did change other errors in that file to OSError.